### PR TITLE
HOCS-3189 Moved to gradle 7.1 and spring boot 2.5.1

### PR DESCRIPTION
**Pull request by gardnersam**

### CODE REVIEW

File: src/test/java/uk/gov/digital/ho/hocs/templates/TemplateServiceIntTest.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public void setUp() throws IOException {
     @Test
     public void shouldReturnPopulatedTemplate() throws Exception {
         ResponseEntity<byte[]> result = testRestTemplate.exchange(
-                getBasePath() + "case/" + CASE_UUID + "/template/" + TEMPLATE_UUID, GET, new HttpEntity(createValidAuthHeaders()), byte[].class);
+                getBasePath() + "/case/" + CASE_UUID + "/template/" + TEMPLATE_UUID, GET, new HttpEntity(createValidAuthHeaders()), byte[].class);
```
Comment (robertdeniszczyc2) [inline src/test/java/uk/gov/digital/ho/hocs/templates/TemplateServiceIntTest.java:103]: I can see this change is required as the test fails without the change, but what is the reason the test now breaks? What is the change in gradle/spring which means we can drop the leading slash here?

**Review by robertdeniszczyc2 — COMMENTED** [review]
Just added one question which confused me

File: src/test/java/uk/gov/digital/ho/hocs/templates/TemplateServiceIntTest.java (hunk 1 above)
Comment (gardnersam) [reply src/test/java/uk/gov/digital/ho/hocs/templates/TemplateServiceIntTest.java:103]: I'm not sure on the version, but on both the string created for the URL looks like this "http://localhost:53021case/11111111-1111-1111-1111-111111111111/template/22222222-2222-2222-2222-222222222222" Up to date spring boot code points out that 53021case is not a port whereas previous versions seem to let you build a URL with a nonsense port. So in this case I think the test was wrong, but Spring was letting it slide

**Review by robertdeniszczyc2 — APPROVED** [review]
Confirmed builds and runs locally

**Pull request merged**